src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -266,22 +318,39 @@ public void Save(ModelSaveContext ctx)
             _host.AssertValue(ctx);
             ctx.CheckAtModel();
             ctx.SetVersionInfo(GetVersionInfo());
+            ctx.Writer.Write(IsFrozen ? 1 : 0);
+
             // *** Binary format ***
+            // int: indicator for frozen models
             // stream: tensorFlow model.
             // int: number of input columns
             // for each input column
             //   int: id of int column name
             // int: number of output columns
             // for each output column
             //   int: id of output column name
+            if (IsFrozen)
+            {
+                var buffer = new TFBuffer();
+                Session.Graph.ToGraphDef(buffer);
 
-            var buffer = new TFBuffer();
-            Session.Graph.ToGraphDef(buffer);
-
-            ctx.SaveBinaryStream("TFModel", w =>
+                ctx.SaveBinaryStream("TFModel", w =>
+                {
+                    w.WriteByteArray(buffer.ToArray());
+                });
+            }
+            else
             {
-                w.WriteByteArray(buffer.ToArray());
-            });
+                var tfSavedModelDirectoryInfo = new TensorFlowSavedModelDirectoryInfo();
+                var byteArray = tfSavedModelDirectoryInfo.PackageTensorFlowSavedModel(_model);
```

Comment:
>_model [](start = 86, length = 6)

for un-frozen models we need to save the entire SavedModel folder  structure. Here I am :

(a) packaging the folder structure and creating a byte[]
(b)  Saving the byte array when  calling ML.NET Save()  
(3) cleaning up the temporary  directory  used for doing (a)